### PR TITLE
MMDLoader: Apply color management to textures.

### LINKED ISSUE #26553: MMDLoader Colorization Issue
### Description

When using recent versions of Three.js (155) the MMDLoader appears to be loading PMX files with washed out colors. Using the code from the following link: [three.js webgl - loaders - MMD loader](https://threejs.org/examples/webgl_loader_mmd.html) , the Miku model renders fine (this is an older PMD file with simpler textures), but trying it against another, newer PMX file the colors appear very washed out. See the following result:

image

It seems that the colorization on MMD files has been broken since r92. Loading the file using the older r92 Three.js code works fine (see the following):

image

### Reproduction steps

1. Use MMDLoader for example using the code from the following MMDLoader example: https://threejs.org/examples/webgl_loader_mmd.html
2. Replace the **PMD** file referenced within the code with any **PMX** file
3. Observe the results

### Code

```
const modelFile = 'models/mmd/miku/miku_v2.pmd';  **// Replace this with any .pmx file**
const vmdFiles = [ 'models/mmd/vmds/wavefile_v2.vmd' ];

```

### Live example

https://jsfiddle.net/jasondalycan/cum250ad/1/

### Screenshots

_No response_

### Version

r155

### Device

Desktop

### Browser

Chrome, Edge

### OS

Windows

### PR DESCRIPTION
**Pull request by donmccurdy**
- Fixes #26553
- Related #26479
- Related #26135
- Related #25889

| before | after |
|---|---|
| before | after |

### CODE REVIEW

File: examples/jsm/loaders/MMDLoader.js [hunk 1]
```diff
@@ -1133,17 +1134,18 @@ class MaterialBuilder {
 				 * MMDToonMaterial doesn't have ambient. Set it to emissive instead.
 				 * It'll be too bright if material has map texture so using coef 0.2.
 				 */
-			params.diffuse = new Color().fromArray( material.diffuse );
+			params.diffuse = new Color().setRGB(
+				material.diffuse[ 0 ],
+				material.diffuse[ 1 ],
+				material.diffuse[ 2 ],
+				SRGBColorSpace
+			);
```
Comment (mrdoob) [inline examples/jsm/loaders/MMDLoader.js:1142]: Why not `params.diffuse = new Color().setRGB( ...material.diffuse, SRGBColorSpace );` here too?

**Review by takahirox — APPROVED** [review]
Good catch, yes we are missing color space assignment on color textures in MMD loader.

**Comment by takahirox** [comment]
I tested locally with a PMX file. Using SRGB color space looks fine.

Reference:

image

|before|after (SRGB)|LinearSRGB|
|----|----|----|
|image|image|image|

File: examples/jsm/loaders/MMDLoader.js (hunk 1 above)
Comment (donmccurdy) [reply examples/jsm/loaders/MMDLoader.js:1142]: The issue here is the 4th component in the diffuse array, otherwise it would be fine.

**Comment by jasondalycan** [comment]
Thanks guys.  It looks better (more color now) but it still looks a bit light compared to the rendering in r92.  Not sure if this is the intention?  See the following:

r92:
image

After this commit (26554):
image

**Comment by donmccurdy** [comment]
@jasondalycan the renderings above appear similar enough to me that they may be caused only by lighting and not color management. Some adjustments to lighting are to be expected, given changes in the past few releases. Adjustment to `renderer.toneMapping` or `renderer.toneMappingExposure` may be worth trying as well, but less likely in this case.

If there does seem to be a remaining color management issue, I think we would need an example (model) to investigate it.

**Comment by jasondalycan** [comment]
Ok thanks I'll work with adjusting lighting settings and report back if any additional findings.  cheers

**Comment by danpaldev** [comment]
> Ok thanks I'll work with adjusting lighting settings and report back if any additional findings. cheers

By any chance did you find a fix for this? Almost a year later I keep observing the washed out colors, even with lighting on/off.

Threejs
image

Other project:
image

**Comment by Mugen87** [comment]
see https://github.com/mrdoob/three.js/issues/28336#issuecomment-2105643560

**Pull request merged**